src/hw/move_zeroes/Solution.java

```diff
@@ -0,0 +1,34 @@
+package hw.move_zeroes;
+
+import java.util.Arrays;
+
+/// Given an integer array numbers, move all 0's to the end of it while maintaining the relative order of the non-zero elements.
+///
+/// Note that you must do this in-place without making a copy of the array.
+///
+/// @link [Move Zeroes](https://leetcode.com/problems/move-zeroes/)
+public class Solution {
+    /// Main.
+    ///
+    /// @param args Arguments list.
+    public static void main(String[] args) {
+        int[] numbers = {2, 4, 0, 7, 0, 0, 0, 8, 9};
+        moveZeroes(numbers);
+        System.out.println(Arrays.toString(numbers));
+    }
+
+    /// Moves zeros to the end with preserving order.
+    ///
+    /// @param numbers Array to process.
+    public static void moveZeroes(int[] numbers) {
+        int newPosition = 0;
+        for (int j = 0; j < numbers.length; j++) {
+            if (numbers[j] == 0)
+                continue;
```

Comment:
Please configure checkstyle in IDE and pom.xml
```suggestion
            if (numbers[j] == 0) {
                continue;
            }
```